framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -418,19 +420,42 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
-            if (postcondAnno == null) {
-                continue;
-            }
-            for (String expr : expressions) {
-                result.add(new Postcondition(expr, postcondAnno, anno));
+            /**
+             * If the {@code value} field is a list of Strings, then the try block will work
```

Comment:
In a previous code review, I said not to use exceptions as control flow.  You could write an explicit check of the `value` field, but don't make the code throw an exception when an `if` statement is possible.

Reply:
@mernst I tried but was not able to. How to check that an `AnnotationMirror anno` consists of String class or AnnotationMirror class?

Reply:
Can you tell me what you tried?  Hint: there are only two methods in the [`AnnotationMirror` class](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/element/AnnotationMirror.html).

In general, don't just say "I tried but was not able to".  That is indistinguishable from you doing nothing at all, and it does not enable me to help you.
Rather than asking me to do all the work, it is better for you to say the specific part of it you are having trouble with.

Reply:
@mernst I finally worked out a way on how to remove try-catch. I have made a commit related to it, please do review it. I will push the commit related to the test cases in a nick of time.